src/ripple/ledger/impl/ApplyStateTable.cpp

```diff
@@ -0,0 +1,589 @@
+                    meta.getAffectedNode (item.first).emplace_back (std::move(finals));
+            }
+            else if (type == &sfModifiedNode)
+            {
+                assert (curNode && origNode);
+
+                if (curNode->isThreadedType ()) // thread transaction to node item modified
+                    threadTx (meta, curNode, newMod);
+
+                STObject prevs (sfPreviousFields);
+                for (auto const& obj : *origNode)
+                {
+                    // search the original node for values saved on modify
+                    if (obj.getFName ().shouldMeta (SField::sMD_ChangeOrig) && !curNode->hasMatchingEntry (obj))
+                        prevs.emplace_back (obj);
+                }
+
+                if (!prevs.empty ())
+                    meta.getAffectedNode (item.first).emplace_back (std::move(prevs));
+
+                STObject finals (sfFinalFields);
+                for (auto const& obj : *curNode)
+                {
+                    // search the final node for values saved always
+                    if (obj.getFName ().shouldMeta (SField::sMD_Always | SField::sMD_ChangeNew))
+                        finals.emplace_back (obj);
+                }
+
+                if (!finals.empty ())
+                    meta.getAffectedNode (item.first).emplace_back (std::move(finals));
+            }
+            else if (type == &sfCreatedNode) // if created, thread to owner(s)
+            {
+                assert (curNode && !origNode);
+                threadOwners (to, meta, curNode, newMod, j);
+
+                if (curNode->isThreadedType ()) // always thread to self
+                    threadTx (meta, curNode, newMod);
+
+                STObject news (sfNewFields);
+                for (auto const& obj : *curNode)
+                {
+                    // save non-default values
+                    if (!obj.isDefault () &&
+                            obj.getFName().shouldMeta(
+                                SField::sMD_Create | SField::sMD_Always))
+                        news.emplace_back (obj);
+                }
+
+                if (!news.empty ())
+                    meta.getAffectedNode (item.first).emplace_back (std::move(news));
+            }
+            else
+            {
+                assert (false);
+            }
+        }
+
+        // add any new modified nodes to the modification set
+        for (auto& mod : newMod)
+            update (to, mod.second);
+
+        sMeta = std::make_shared<Serializer>();
+        meta.addRaw (*sMeta, ter, to.txCount());
+
+        // VFALCO For diagnostics do we want to show
+        //        metadata even when the base view is open?
+        JLOG(j.trace) <<
+            "metadata " << meta.getJson (0);
+    }
+    to.rawTxInsert(
+        tx.getTransactionID(),
+            sTx, sMeta);
+    apply(to);
+}
+
+//---
+
+bool
+ApplyStateTable::exists (ReadView const& base,
+    Keylet const& k) const
```

Comment:
I believe that the function is correct as written, but if you think there's something missing I'd love to hear it